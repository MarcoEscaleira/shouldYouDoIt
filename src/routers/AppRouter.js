import React, { Fragment } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import HomePage from '../pages/HomePage';
import GalleryPage from '../pages/GalleryPage';
import Header from '../components/Header';
import Footer from '../components/Footer';
import NotFound from '../components/NotFound';

const AppRouter = () => (
  <BrowserRouter>
    <Fragment>
      <Header />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/gallery" component={GalleryPage} />
        <Route component={NotFound} />
      </Switch>
      <Footer />
    </Fragment>
  </BrowserRouter>
);

export default AppRouter;