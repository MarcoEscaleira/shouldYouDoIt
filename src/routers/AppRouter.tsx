import React, { Fragment } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import HomePage from "~/pages/HomePage/HomePage";
import GalleryPage from "~/pages/GalleryPage/GalleryPage";
import Header from "~/components/Header/Header";
import Footer from "~/components/Footer/Footer";
import NotFound from "~/components/NotFound/NotFound";

import { AppContainer } from "./styled-components";

const AppRouter: React.FC = () => (
  <BrowserRouter>
    <Fragment>
      <AppContainer>
        <Header />
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/gallery" component={GalleryPage} />
          <Route component={NotFound} />
        </Switch>
      </AppContainer>
      <Footer />
    </Fragment>
  </BrowserRouter>
);

export default AppRouter;
