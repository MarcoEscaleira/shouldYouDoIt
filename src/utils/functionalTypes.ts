export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
