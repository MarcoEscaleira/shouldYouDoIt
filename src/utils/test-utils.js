import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
