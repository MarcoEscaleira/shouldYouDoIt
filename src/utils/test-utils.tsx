import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
