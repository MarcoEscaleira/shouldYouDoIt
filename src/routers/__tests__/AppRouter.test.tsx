import React from "react";
import AppRouter from "../AppRouter";
import { renderWithRedux } from "~/utils/test-utils";

describe("App Router test suite", () => {
  it("Should render App Router correctly", () => {
    const { container } = renderWithRedux(<AppRouter />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
