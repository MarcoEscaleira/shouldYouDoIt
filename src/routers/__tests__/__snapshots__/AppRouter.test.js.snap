// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Router test suite Should render App Router correctly 1`] = `
<div
  class="main"
>
  <header
    class="header"
  >
    <nav
      class="header__nav"
    >
      <a
        aria-current="page"
        class="nav__title active"
        href="/"
      >
        Should You 
        <span
          class="nav__title__span"
        >
          Do It
        </span>
      </a>
      <ul
        class="nav__ul"
      >
        <li
          class="nav__ul__li"
        >
          <a
            aria-current="page"
            class="navLink isActive"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav__ul__li"
        >
          <a
            class="navLink"
            href="/gallery"
          >
            Gallery
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    class="container"
  >
    <div
      class="questionForm"
    >
      <h1
        class="questionForm__title"
      >
        What do you want to do?
      </h1>
      <form
        autocomplete="off"
        class="questionForm__form"
      >
        <input
          class="questionForm__form__input"
          data-testid="input"
          name="toDo"
          type="text"
          value=""
        />
        <button
          class="questionForm__form__btn"
          type="submit"
        >
          Decide
        </button>
      </form>
    </div>
    <div
      class="modal modal--invisible"
    >
      <div
        class="modal__overlay"
      >
        <div
          class="modal__content"
        >
          <h3
            class="modal__title"
          >
            Error
          </h3>
          <p
            class="modal__body"
          />
          <button
            class="modal__button"
          >
            Okay
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
